src/Image.cc

```diff
@@ -218,6 +218,14 @@ Image::loadFromBuffer(uint8_t *buf, unsigned len) {
   }
 #endif
 #endif
+#ifdef HAVE_RSVG
+  // confirm svg using first 1000 chars
+  // if a very long comment precedes the root <svg> tag, isSVG returns false
+  uint8_t head[1000] = {0};
+  unsigned head_len = (len < 1000 ? len : 1000);
+  memcpy(head, buf, head_len * sizeof(uint8_t));
+  if (isSVG(head, head_len)) return loadSVGFromBuffer(buf, len);
```

Comment:
Is it possible to directly pass `buf` to `isSVG` here? I mean `head_len` should already specify the max length, so one extra `memcpy` could be skipped. ;)